innopoints/views/application.py

```diff
@@ -0,0 +1,79 @@
+"""Views related to the Application model.
+
+- POST /projects/{project_id}/activity/{activity_id}/apply
+- DELETE /projects/{project_id}/activity/{activity_id}/apply
+"""
+
+import logging
+
+from flask import abort, request
+from flask_login import login_required, current_user
+from sqlalchemy.exc import IntegrityError
+
+from innopoints.extensions import db
+from innopoints.blueprints import api
+from innopoints.models import Application, Project, Activity, ApplicationStatus
+from innopoints.schemas import ApplicationSchema
+
+
+NO_PAYLOAD = ('', 204)
+log = logging.getLogger(__name__)
+
+@api.route('/projects/<int:project_id>/activity/<int:activity_id>/apply', methods=['POST'])
+@login_required
+def apply_for_activity(project_id, activity_id):
+    """Apply for volunteering on a particular activity."""
+    if not request.is_json:
+        abort(400, {'message': 'The request should be in JSON.'})
+
+    project = Project.query.get_or_404(project_id)
+    activity = Activity.query.get_or_404(activity_id)
+    if activity.project != project:
+        abort(400, {'message': 'The specified project and activity are unrelated.'})
+
+    if activity.has_application_from(current_user):
+        abort(400, {'message': 'An application already exists.'})
+
+    if activity.telegram_required and not isinstance(request.json.get('telegram'), str):
```

Comment:
Perhaps should check that it's a valid username? Just that it matches the regex `\w+` for example

Reply:
Well, we don't exactly care if it's valid or not, this check will exist in the front-end, but if someone thinks they're smart, it will be against their interests.

Reply:
Never rely on the frontend. It is the easiest thing to "hack", and this check is very simple: just `\w{5,32}`

Reply:
I get it, but as I said, there is nothing to hack. We don't care what this value is, we don't use it anywhere except in display. I think this check will be excess code